app/classes/document_search_results.rb

```diff
@@ -30,18 +31,35 @@ def extract_hits(hits)
       source =  deserialized(hit)
       if highlight.present?
         source['title'] = highlight["title_#{source['language']}"].first if highlight["title_#{source['language']}"]
-        %w(description content).each do |optional_field|
+        %w[description content].each do |optional_field|
           language_field = "#{optional_field}_#{source['language']}"
           source[optional_field] = highlight[language_field].join('...') if highlight[language_field]
         end
       end
-      %w(created_at created changed updated_at updated).each do |date|
+      %w[created_at created changed updated_at updated].each do |date|
         source[date] = Time.parse(source[date]).utc.to_s if source[date].present?
       end
       source
     end
   end
 
+  def extract_aggregations(aggregations)
+    return unless aggregations
+
+    aggregations.filter_map do |field, data|
+      if data['buckets'].present?
+        { "#{field}": extract_aggregation_rows(data['buckets']) }
+      end
+    end
+  end
+
+  def extract_aggregation_rows(rows)
+    rows.map do |term_hash|
+      { value: term_hash['key'],
```

Comment:
Digging into this a bit more, the returned hash is raising an eyebrow. We're converting their original `key` key to `value`, which could be a source of confusion:
```ruby
# raw ES result:
          "tags" => {
        "doc_count_error_upper_bound" => 0,
                "sum_other_doc_count" => 0,
                            "buckets" => [
            [0] {
                      "key" => "tag1",
                "doc_count" => 1
            },
            [1] {
                      "key" => "tag2",
                "doc_count" => 1
            }
        ]

# our extracted hash
{
        :tags => [
            {
                    :value => "tag1",
                :doc_count => 1
            },
            {
                    :value => "tag2",
                :doc_count => 1
            }
        ]
    }
```
We should either preserve their original `key` key for consistency, or even use the key itself as the key to the doc count, i.e.:
```ruby
# ignore the 1.9 hash syntax - just copied and pasted
{
        :tags => [
           {
                   "tag1" => 1
            },
             {
                   "tag2" => 1
            }
        ]
    }
```
There could be ramifications to the latter simplification that I'm not seeing.

Reply:
@MothOnMars We can discuss more, but my preference is to leave this as is for two reasons.  

1. The variable name change from `key` to `value` (or whatever, I'm not particularly wedded to value per se) allows us to call up this value with the `agg.value` shorthand without running into the conflict that `key` presents since it is also a hash function.  (`agg.key` will return a `wrong number of arguments (given 0, expected 1)` error because it is assumed we are calling the `key` function, not accessing key.)  Sure, we can accomplish the same with `agg['key']`, but it just doesn't seem as clean.  I'm willing to be over-ruled, but `key` just strikes me as "special-ish" and something that shouldn't be persisted if there's no reason not to.

2. I agree the second recommendation is nicer/cleaner, but that structure is only sensible for these text-based aggregations.  When we move over to dates we'll need to return a hash that has more than just key/value and doc_count.  It will instead look more like:

```
      { value: term_hash['key'],
        doc_count: term_hash['doc_count'],
        to: term_hash['to'],
        from: term_hash['from'],
        to_as_string: term_hash['to_as_string'],
        from_as_string: term_hash['from_as_string'] }
```

So, while this structure seems overly complex when we only have key/doc_count, it lends itself to more gracefully mirroring what we'll need with more complicated aggregations.

Reply:
Sorry for the delayed reply.

I'd prefer to use something other than `value`, as that also is a special term. I don't have any great alternative names. `agg_key`?